Platform/ApollolakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -578,9 +578,11 @@ ClearFspHob (
 
   LdrGlobal  = (LOADER_GLOBAL_DATA *)GetLoaderGlobalDataPointer ();
   HandOffHob = (EFI_HOB_HANDOFF_INFO_TABLE  *) LdrGlobal->FspHobList;
-  Length     = (UINT8 *) (UINTN) HandOffHob->EfiEndOfHobList - (UINT8 *)HandOffHob;
-  ZeroMem (HandOffHob, Length);
-  LdrGlobal->FspHobList = NULL;
+  if (HandOffHob != NULL) {
```

Comment:
Do we know why FspHobList could be NULL? we might need update logic to make sure only free it once.

Reply:
FspHobClear() will be called in board notification.  
And FspHobList will be set to NULL.   When in OsLoader, it tries to notify again,  this will happen.

Reply:
Could we also avoid to call FspHobClear() twice?

Reply:
I think this patch will handle it.  When FspHobClear() is called more than once, it will just return.

Reply:
Agree this patch could WA the issue.  At same time, should we avoid it is invoked twice?

Reply:
I think it is better to handle it inside ClearHobClear() to make it work in either way.
ClearHobClear() will be called from board notification.  The board notification should be called only once per notify code.   This is not enforced for now.

Reply:
Some notifications, e.g. EndOfStages, EndOfFirmware should be invoked only once.
I am ok to check this fix firstly and update others code later.

Reply:
OK, let us handle it separately.
For this fix, it is required anyway since it can potentially access NULL pointer.